ioc/start.sh

```diff
@@ -120,14 +120,39 @@ if [ -f ${override} ]; then
 # 2. ioc.yaml ******************************************************************
 elif [ -f ${ibek_src} ]; then
 
-    if [[ ${#ibek_yams[@]} > 1 ]]; then
+    if [[ ${#ibek_yamls[@]} > 1 ]]; then
         ibek_error "ERROR: Multiple YAML files found in ${CONFIG_DIR}."
     fi
 
     # Database generation script generated by ibek
     db_src=${RUNTIME_DIR}/ioc.subst
     final_ioc_startup=${RUNTIME_DIR}/st.cmd
 
+    # Find address (ID) and CLASS of the first camera in the IOC instance yaml.
+    # NOTE this does not support mixed types of cameras in the same IOC
+    # ONE camera per IOC instance is recommended
+    # Multiple cameras of the same type are supported
+    # Mixed cameras will work but may not expose all their GenICam config parameters
+    instance_class=$(grep -m1 -oP "(?<=CLASS:\s).*" ${ibek_src}) || true  # https://regex101.com/r/358gq3/1
```

Comment:
Could we use [yq](https://github.com/mikefarah/yq) for this?

Reply:
Yes we are now using yq and that has allowed Marcell to support looping over all the cameras and creating individual pvi db and bob for each.